fh-sync.d.ts

```diff
@@ -0,0 +1,407 @@
+          metricsReportConcurrency?: number;
+          /** If cache the dataset client records using redis. This can help improve performance for the syncRecords API.
+           * Can be turned on if there are no records are shared between many different dataset clients. Default is false.*/
+          useCache?: boolean;
+          /**The TTL (Time To Live) value for the messages on the queue. In seconds. Default to 24 hours. */
+          queueMessagesTTL?: string;
+          /** Specify the maximum retention time of an inactive datasetClient. Any inactive datasetClient that is older than this period of time will be removed.*/
+          datasetClientCleanerRetentionPeriod?: string;
+          /** Specify the frequency the datasetClient cleaner should run. Default every hour ('1h').*/
+          datasetClientCleanerCheckFrequency?: string;
+      }
+
+      /**
+       * Backoff Strategy
+       * Example: {strategy: 'exp', max: 60*1000},
+       */
+      interface PendingWorkerBackoff {
+          strategy: string;
+          max: number;
+      }
+      /**
+       * Backoff Strategy
+       * Example: {strategy: 'exp', max: 60*1000},
+       */
+      interface AckWorkerBackoff {
+          strategy: string;
+          max: number;
+      }
+
+      /**
+       * Backoff Strategy
+       * Example: {strategy: 'exp', max: 60*1000},
+       */
+      interface SyncWorkerBackoff {
+          strategy: string;
+          max: number;
+      }
+
+      type StandardCb<T> = (err: Error | null | string | undefined, res?: T | undefined) => void;
+      type NoRespCb = (err: Error | string | undefined) => void;
+
+      /**
+       * Options used to initialize sync for specific dataset
+       */
+      interface SyncInitOptions {
+          /**
+           * Value indicating how often the dataset client should be sync with the backend. Matches the clients default
+           * frequency. Value in seconds
+           */
+          syncFrequency?: number,
+
+          /**
+           * Value that will be used to decide if the dataset client is not active anymore.
+           */
+          clientSyncTimeout?: number,
+
+          /**
+           * Value that determines how long it should wait for the backend list operation to complete
+           */
+          backendListTimeout?: number,
+
+          /**
+           * Specify the max wait time the dataset can be scheduled to sync again after its previous schedule, in seconds.
+           */
+          maxScheduleWaitTime?: number
+      }
+
+      /**
+       * Parameters object for request and response interceptors
+       */
+      interface SyncInterceptParams {
+          query_params: any;
+          metaData: any;
+      }
+      /**
+       * Connect sync server to mongo and redis. Returns the MongoDB and Redis clients being used internally.
+       *
+       * @param mongoDBConnectionUrl - Unique id of the dataset (usually collection, table in your database)
```

Comment:
Function comments are now uniform and display a little more consistently. Should we add more detail in comments?